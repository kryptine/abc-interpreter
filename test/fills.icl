module fills

:: *UT
	= UT10 Int
	| UT11 !Int
	| UT201 Int !Int
	| UT211 !Int !Int

f :: !Bool !*UT -> *(!Bool, !*UT)
f True  (UT10 a) = (True,  UT11 a)
f True  (UT11 a) = (True,  UT10 a)
f False (UT10 a) = (False, UT10 a)
f False (UT11 a) = (True,  UT11 a)
f True  (UT201 a b) = (True,  UT211 b a)
f False (UT201 a b) = (False, UT201 a a)
f True  (UT211 a b) = (False, UT201 b a)
f False (UT211 a b) = (True,  UT211 b b)

Start =
	[ f True  (UT10 1)
	, f False (UT10 1)
	, f True  (UT11 1)
	, f False (UT11 1)
	, f True  (UT201 1 2)
	, f False (UT201 1 2)
	, f True  (UT211 1 2)
	, f False (UT211 1 2)
	]
