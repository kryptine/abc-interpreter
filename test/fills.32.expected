[(True,(UT2 5)),(True,(UT1 10)),(False,(UT1 15)),(True,(UT2 20))]
halt at 6
56 524232 524288
